aws/data_source_aws_iam_policy_document.go

```diff
@@ -114,14 +115,22 @@ func dataSourceAwsIamPolicyDocumentRead(d *schema.ResourceData, meta interface{}
 	if cfgStmts, hasCfgStmts := d.GetOk("statement"); hasCfgStmts {
 		var cfgStmtIntf = cfgStmts.([]interface{})
 		stmts := make([]*IAMPolicyStatement, len(cfgStmtIntf))
+		sidMap := make(map[string]struct{})
+
 		for i, stmtI := range cfgStmtIntf {
 			cfgStmt := stmtI.(map[string]interface{})
 			stmt := &IAMPolicyStatement{
 				Effect: cfgStmt["effect"].(string),
 			}
 
 			if sid, ok := cfgStmt["sid"]; ok {
+				if _, ok := sidMap[sid.(string)]; ok {
+					return fmt.Errorf("Sid should be blank or unique. Change Sid : %s", sid.(string))
```

Comment:
A few things here:
* Can you please add an acceptance test that covers this new behavior by matching against this error (e.g. `ExpectError: regexp.MustCompile()`)
* I would suggest updating the wording slightly here:

```suggestion
					return fmt.Errorf("Found duplicate sid (%s). Either remove the sid or ensure the sid is unique across all statements.", sid.(string))
```